### PR TITLE
Jv work with file - implemented getStatistic method

### PR DESCRIPTION
**Pull request by olieniev**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,42 @@
...
+        int result = supplyCount - buyCount;
+        builder.append("supply,").append(supplyCount).append(System.lineSeparator());
+        builder.append("buy,").append(buyCount).append(System.lineSeparator());
+        builder.append("result,").append(result).append(System.lineSeparator());
+        File toFile = new File(toFileName);
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFile))) {
+            bufferedWriter.write(builder.toString());
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write to file", e);
+        }
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/WorkWithFile.java:40]: enormous big method, that violates SRP principle. should be refactored
```suggestion
        String dataFromFile = readFromFile(fromFileName);
        String report = processData(dataFromFile);
        writeToFile(report, toFileName);
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,42 @@
...
+                String[] lineValues = fileString.split(",");
+                if (lineValues[0].equals("supply")) {
+                    supplyCount += Integer.parseInt(lineValues[1]);
+                } else {
+                    buyCount += Integer.parseInt((lineValues[1]));
+                }
+                fileString = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file", e);
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/WorkWithFile.java:28]: specify filename with caused error

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,7 +1,42 @@
...
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
-
+        File fromFile = new File(fromFileName);
+        int supplyCount = 0;
+        int buyCount = 0;
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(fromFile));
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/WorkWithFile.java:16]: creating File for that is not neccesary
```suggestion
        int supplyCount = 0;
        int buyCount = 0;
        try {
            BufferedReader reader = new BufferedReader(new FileReader(fromFileName));
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,7 +1,42 @@
...
+        StringBuilder builder = new StringBuilder();
+        int result = supplyCount - buyCount;
+        builder.append("supply,").append(supplyCount).append(System.lineSeparator());
+        builder.append("buy,").append(buyCount).append(System.lineSeparator());
+        builder.append("result,").append(result).append(System.lineSeparator());
+        File toFile = new File(toFileName);
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFile))) {
+            bufferedWriter.write(builder.toString());
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write to file", e);
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/WorkWithFile.java:39]: specify filename with caused error

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -1,7 +1,42 @@
...
+        int buyCount = 0;
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(fromFile));
+            String fileString = reader.readLine();
+            while (fileString != null) {
+                String[] lineValues = fileString.split(",");
+                if (lineValues[0].equals("supply")) {
+                    supplyCount += Integer.parseInt(lineValues[1]);
+                } else {
+                    buyCount += Integer.parseInt((lineValues[1]));
```
Comment (kerrrusha) [inline src/main/java/core/basesyntax/WorkWithFile.java:23]: 0, 1 - magic values, should be constants

**Review by kerrrusha — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 6]
```diff
@@ -1,7 +1,55 @@
...
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    static final int OPERATION_TYPE_POSITION = 0;
+    static final int AMOUNT_POSITION = 1;
```
Comment (sokimaaa) [inline src/main/java/core/basesyntax/WorkWithFile.java:11]: make constant private
```suggestion
    private static final int OPERATION_TYPE_POSITION = 0;
    private static final int AMOUNT_POSITION = 1;
```

**Review by sokimaaa — APPROVED** [review]

**Pull request closed without merging**